test_conformance/images/kernel_read_write/test_cl_ext_image_from_buffer.cpp

```diff
@@ -79,77 +79,69 @@ int image2d_from_buffer_positive(cl_device_id device, cl_context context,
         return TEST_SKIPPED_ITSELF;
     }
 
-    std::vector<cl_mem_object_type> imageTypes{
-        CL_MEM_OBJECT_IMAGE1D,       CL_MEM_OBJECT_IMAGE2D,
-        CL_MEM_OBJECT_IMAGE3D,       CL_MEM_OBJECT_IMAGE1D_BUFFER,
-        CL_MEM_OBJECT_IMAGE1D_ARRAY, CL_MEM_OBJECT_IMAGE2D_ARRAY
-    };
+    cl_mem_object_type imageType = CL_MEM_OBJECT_IMAGE2D;
 
     std::vector<cl_mem_flags> flagTypes{ CL_MEM_READ_ONLY, CL_MEM_WRITE_ONLY,
                                          CL_MEM_READ_WRITE,
                                          CL_MEM_KERNEL_READ_AND_WRITE };
 
     for (auto flagType : flagTypes)
     {
-        for (auto imageType : imageTypes)
+
+        /* Get the list of supported image formats */
+        std::vector<cl_image_format> formatList;
+        if (TEST_PASS
+                != get_format_list(context, imageType, formatList, flagType)
+            || formatList.size() == 0)
         {
-            /* Get the list of supported image formats */
-            std::vector<cl_image_format> formatList;
-            if (TEST_PASS
-                    != get_format_list(context, imageType, formatList, flagType)
-                || formatList.size() == 0)
-            {
-                test_fail("Failure to get supported formats list\n");
-            }
+            test_fail("Failure to get supported formats list\n");
+        }
 
-            cl_uint row_pitch_alignment_2d = 0;
-            cl_int err =
-                clGetDeviceInfo(device, CL_DEVICE_IMAGE_PITCH_ALIGNMENT,
-                                sizeof(row_pitch_alignment_2d),
-                                &row_pitch_alignment_2d, nullptr);
-            test_error(err, "Error clGetDeviceInfo");
+        cl_uint row_pitch_alignment_2d = 0;
+        cl_int err = clGetDeviceInfo(device, CL_DEVICE_IMAGE_PITCH_ALIGNMENT,
+                                     sizeof(row_pitch_alignment_2d),
+                                     &row_pitch_alignment_2d, nullptr);
+        test_error(err, "Error clGetDeviceInfo");
 
-            cl_uint base_address_alignment_2d = 0;
-            err =
-                clGetDeviceInfo(device, CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT,
-                                sizeof(base_address_alignment_2d),
-                                &base_address_alignment_2d, nullptr);
-            test_error(err, "Error clGetDeviceInfo");
+        cl_uint base_address_alignment_2d = 0;
+        err = clGetDeviceInfo(device, CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT,
+                              sizeof(base_address_alignment_2d),
+                              &base_address_alignment_2d, nullptr);
+        test_error(err, "Error clGetDeviceInfo");
```

Comment:
Minor, these queries dont' depend on the "flagType", so they could be moved outside of the loop.

Reply:
thank I'll amend the code